src/libambit/libambit.c

```diff
@@ -320,13 +321,27 @@ void libambit_log_entry_free(ambit_log_entry_t *log_entry)
 
 static int device_info_get(ambit_object_t *object, ambit_device_info_t *info)
 {
+    uint8_t *komposti_version;
+    size_t komposti_version_size;
     uint8_t *reply_data = NULL;
     size_t replylen;
     int ret = -1;
 
     LOG_INFO("Reading device info");
 
-    if (libambit_protocol_command(object, ambit_command_device_info, komposti_version, sizeof(komposti_version), &reply_data, &replylen, 1) == 0) {
+    switch (info->product_id) {
```

Comment:
Could this kind of information be in device_support.c? Adding support for new devices would otherwise require changes here & maybe more places, I'm not that familiar with this code yet.
An example: I've got a Ambit3 Vertical (product id 0x002c) .. which doesn't work with openambit.. 
Note, product id etc for Ambit3 Vertical can be found here: https://github.com/kbck/openambit/tree/ambit3_vertical